### PR TITLE
Switch config to yml style.

### PR DESCRIPTION
**Pull request by joshkreud**
Changes static config style to yml. 

Global middleware assignment taken from #7
Middleware disabled in dev mode though. 

The change is breaking old configs!
New Static prod config file must be present when deploying prod.

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,14 +1,4 @@
 # edit and save as .env
-COMPOSE_FILE=docker-compose.base.yml:docker-compose.dev.yml
 TZ=Europe/Berlin
-DOMAINNAME=docker.localhost
```
Comment (hengeb) [inline .env.sample:4]: why? dev configuration might be simpler with this environment variable

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -1,44 +1,24 @@
...
+      - ${VOLUMES_DIR}/log:/var/log
+      - ./conf/traefik-dyn.yml:/etc/traefik/config/dnamic.yml
+      - ./conf/traefik-static.dev.yml:/etc/traefik/traefik.yml
     labels:
       - traefik.enable=true
       - traefik.http.routers.traefik-dashboard.entrypoints=websecure
       - traefik.http.routers.traefik-dashboard.service=api@internal
-      - traefik.http.routers.traefik-dashboard.rule=Host(`traefik.${DOMAINNAME}`)
-      - traefik.http.routers.traefik-dashboard.middlewares=secheader@file
+      - traefik.http.routers.traefik-dashboard.rule=Host(`traefik.docker.localhost`)
```
Comment (hengeb) [inline docker-compose.dev.yml:14]: see comment for `.env`

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -1,44 +1,24 @@
...
+    networks:
+      - traefik
     labels:
       - traefik.enable=true
       - traefik.http.routers.whoami.entrypoints=websecure
-      - traefik.http.routers.whoami.rule=Host(`whoami.${DOMAINNAME}`)
-      - traefik.http.routers.whoami.middlewares=secheader@file
-    networks:
-      - traefik
+      - traefik.http.routers.whoami.rule=Host(`whoami.docker.localhost`)
```
Comment (hengeb) [inline docker-compose.dev.yml:24]: see comment for `.env`

File: README.md [hunk 4]
```diff
@@ -1,23 +1,32 @@
...
+# Traefik
 
-traefik (reverse proxy) setup
+Traefik (reverse proxy) for Mind-Hochschul-Netzwerk
 
-## start
+## Dev setup
 
-Edit env.sample and save it as .env
+Remove ".sample" suffix from [.env](./.env) and change to your needs.
```
Comment (hengeb) [inline README.md]: This could result in an accidentally removed sample file in a git commit.

File: README.md [hunk 5]
```diff
@@ -1,23 +1,32 @@
...
+You will have to repeat this step whenever you restart your container.
 
-    $ docker-compose up -d whoami
+Go to [https://whoami.docker.localhost](https://whoami.docker.localhost) to see if secure connection works. Tell your browser to accept the self-signed certificate.
 
-Go to [http://whoami.docker.localhost](http://whoami.docker.localhost) to see if http connection works.
+## Production setup
 
-Go to [https://whoami.docker.localhost](https://whoami.docker.localhost) to see if secure connection works. Tell your browser to accept the self-signed certificate. You will have to repeat this step whenever you restart your container.
+Remove ".sample" suffix from [.env](./.env) and change to your needs.
```
Comment (hengeb) [inline README.md]: see comment on line 7

File: README.md [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
 
-## production setup
+Remove the "sample" suffix from [traefik-static.prod.yml.sample](./conf/traefik-static.prod.yml.sample) and check the file.
```
Comment (hengeb) [inline README.md]: should be the same workflow as in dev (change requested there)

File: conf/traefik-static.dev.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    http:
+      redirections:
+        entryPoint:
+          to: "websecure"
+          scheme: https
+          permanent: true
+  websecure:
+    address: ":443"
+    http:
+      tls: true
```
Comment (hengeb) [inline conf/traefik-static.dev.yml:35]: right now the secheader@file works fine for me in dev. is that about to change? If not, it should be added here, to have dev closer to prod.

**Review by hengeb — CHANGES_REQUESTED** [review]
see comments

File: .env.sample (hunk 1 above)
Comment (joshkreud) [reply .env.sample:4]: It's only used twice in the dev environment. 
Prod only needs the domainname in the static yml, which can't be templated using the environment. 
Plus in dev it should be "Docker.localhost" in most cases.

File: conf/traefik-static.dev.yml (hunk 7 above)
Comment (joshkreud) [reply conf/traefik-static.dev.yml:35]: I tried it, when applying the headers in dev, I always got an unexplained 404 in both config styles. Removing the headers fixed that. 
I guess hsts is making issues there. 
Maybe limiting the HTTP options may be useful in dev. But apart from that I don't see a reason to keep the headers outside prod.

**Pull request merged**